flink-core/src/main/java/org/apache/flink/configuration/DeploymentOptions.java

```diff
@@ -34,7 +36,16 @@
 			key("execution.target")
 					.stringType()
 					.noDefaultValue()
-					.withDescription("The deployment target for the execution, e.g. \"local\" for local execution.");
+					.withDescription(Description.builder()
+							.text("The deployment target for the execution. This can take one of the following values:")
+							.list(
+									text("remote"),
+									text("local"),
+									text("yarn-per-job"),
+									text("yarn-session"),
+									text("kubernetes-session"))
```

Comment:
I think it would be great to explain how every execution target value is used. This, of course, can happen in the deployment specific documentation but I guess that we should link it then from here.